File: model/application_model.rb
require 'csv'

class ApplicationModel
  def self.read(parser: CSV)
    parser.table("csv/#{filename}.csv")
  end

  def self.find(id)
    all.find { |resource| resource.id == id }
  end

  def self.find_by(column:, value:)
    all.find { |resource| resource.send(column) == value }
  end
end

File: model/application_model.js
import neatCsv from "neat-csv";
import fs from "fs";

export default class ApplicationModel {
  static async read(parser = neatCsv) {
    const csv_file = fs.readFileSync(`csv/${this.filename}.csv`);
    return await parser(csv_file);
  }

  static async find(id) {
    const resources = await this.all();
    return resources.find((resource) => resource.id === id);
  }

  static async find_by(column, value) {
    const resources = await this.all();
    return resources.find((resource) => resource[column] === value);
  }

  static async filter_by(column, value) {
    const resources = await this.all();
    return resources.filter((resource) => resource[column] === value);
  }
}
